### PR TITLE
SQLite storage WIP, tests need to be added

### PR DESCRIPTION
**Pull request by hrishileanplum**
What              | Where/Who
------------------|----------------------------------------
JIRA Issue        | [E2-1988](https://leanplum.atlassian.net/browse/E2-1988)
People Involved   | @gg4race , @e7mac 

## Background
Work in Progress, this gives an idea on the storage , manager and the queue. Work still needs to be done. Especially on API and how the data will be retrieved for multi call etc.

### CODE REVIEW

File: Leanplum/Cache/LPFileManager.m [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+//#import "LPRequestSender.h"
+//#import "LPCountAggregator.h"
+//#import "LPFileTransferManager.h"
+
+typedef enum {
+    kLeanplumFileOperationGet = 0,
+    kLeanplumFileOperationDelete = 1,
+} LeanplumFileTraversalOperation;
+
+NSString *appBundlePath;
```
Comment (e7mac) [inline Leanplum/Cache/LPFileManager.m]: please can we not have global vars?

File: Leanplum/Cache/LPStorageSQLite.m [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+
+#import "LPStorageSQLite.h"
+#import "LPFileManager.h"
+#import "LPConstants.h"
+#import <sqlite3.h>
+
+static sqlite3 *sqlite;
```
Comment (e7mac) [inline Leanplum/Cache/LPStorageSQLite.m]: same here. please make this a local property

Comment (hrishileanplum) [reply Leanplum/Cache/LPStorageSQLite.m]: same as above. We can definitely split this logic, but we will need to investigate how to use these shared values. May be a PR for this to split will help.

File: Leanplum/Manager/LPRequestManager.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+//  Created by Hrishikesh Amravatkar on 6/18/19.
+//  Copyright © 2019 Leanplum. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "LPRequest.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface LPRequestManager : NSObject
```
Comment (e7mac) [inline Leanplum/Manager/LPRequestManager.h]: can we call this `LPRequestStorageManager` or `LPRequestSQLiteStorageManager` `LPRequestSQLiteAdapter`?

File: Leanplum/Api/LPWSManager.m [hunk 4]
```diff
@@ -79,19 +72,22 @@ - (NSString *)generateEncodedQueryString:(NSDictionary *)params withAction:(NSSt
...
-    if ([action isEqualToString:LP_METHOD_MULTI]) {
-        NSString *dataKeyParamString = [NSString stringWithFormat:@"%@=%@", LP_PARAM_DATA,[[self generateEncodedDataString: action] urlencode]];
-        [queryString appendString:dataKeyParamString];
+    NSString *sdkVersionKeyParamString = [NSString stringWithFormat:@"%@=%@&", LP_PARAM_SDK_VERSION,[LPApiConstants sharedState].sdkVersion];
+    [queryString appendString:sdkVersionKeyParamString];
+    NSString *clientParamString = [NSString stringWithFormat:@"%@=%@&", LP_PARAM_CLIENT,[LPApiConstants sharedState].client];
+    [queryString appendString:clientParamString];
+    NSString *timeKeyParamString = [NSString stringWithFormat:@"%@=%@&", LP_PARAM_TIME,[NSString stringWithFormat:@"%f", [[NSDate date] timeIntervalSince1970]]];
+    [queryString appendString:timeKeyParamString];
+    if (params != nil) {
```
Comment (e7mac) [inline Leanplum/Api/LPWSManager.m]: Could we be more obj-c and not do `!= nil` ?

**Pull request merged**